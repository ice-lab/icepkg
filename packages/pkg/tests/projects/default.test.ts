import {runProjectTest} from "./helper";

runProjectTest('default', [
  {
    name: 'default',
    config: {},
    snapshot: 'structure'
  },
  {
    name: 'bundle',
    snapshot: 'structure',
    config: {
      transform: { formats: [] },
      bundle: {}
    }
  },
  {
    name: 'bundle-full',
    snapshot: 'structure',
    config: {
      transform: { formats: [] },
      bundle: {
        formats: ['cjs', 'es2017', 'esm', 'umd']
      }
    }
  },
  {
    name: 'sourcemap-enable',
    snapshot: 'structure',
    config: {
      sourceMaps: true
    }
  }
])
