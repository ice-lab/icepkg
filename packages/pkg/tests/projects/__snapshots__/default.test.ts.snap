// Vitest Snapshot v1

exports[`Run config bundle > cjs structure 1`] = `null`;

exports[`Run config bundle > dist structure 1`] = `
{
  "files": [
    {
      "name": "index.esm.es2017.production.js",
    },
    {
      "name": "index.esm.es5.production.js",
    },
  ],
  "name": "dist",
}
`;

exports[`Run config bundle > es2017 structure 1`] = `null`;

exports[`Run config bundle > esm structure 1`] = `null`;

exports[`Run config bundle-full > cjs structure 1`] = `null`;

exports[`Run config bundle-full > dist structure 1`] = `
{
  "files": [
    {
      "name": "index.cjs.es2017.production.js",
    },
    {
      "name": "index.cjs.es5.production.js",
    },
    {
      "name": "index.esm.es2017.production.js",
    },
    {
      "name": "index.esm.es5.production.js",
    },
    {
      "name": "index.umd.es2017.production.js",
    },
    {
      "name": "index.umd.es5.production.js",
    },
  ],
  "name": "dist",
}
`;

exports[`Run config bundle-full > es2017 structure 1`] = `null`;

exports[`Run config bundle-full > esm structure 1`] = `null`;

exports[`Run config default > cjs structure 1`] = `null`;

exports[`Run config default > dist structure 1`] = `null`;

exports[`Run config default > es2017 structure 1`] = `
{
  "files": [
    {
      "name": "index.d.ts",
    },
    {
      "name": "index.js",
    },
  ],
  "name": "es2017",
}
`;

exports[`Run config default > esm structure 1`] = `
{
  "files": [
    {
      "name": "index.d.ts",
    },
    {
      "name": "index.js",
    },
  ],
  "name": "esm",
}
`;

exports[`Run config sourcemap-enable > cjs structure 1`] = `null`;

exports[`Run config sourcemap-enable > dist structure 1`] = `null`;

exports[`Run config sourcemap-enable > es2017 structure 1`] = `
{
  "files": [
    {
      "name": "index.d.ts",
    },
    {
      "name": "index.js",
    },
    {
      "name": "index.js.map",
    },
  ],
  "name": "es2017",
}
`;

exports[`Run config sourcemap-enable > esm structure 1`] = `
{
  "files": [
    {
      "name": "index.d.ts",
    },
    {
      "name": "index.js",
    },
    {
      "name": "index.js.map",
    },
  ],
  "name": "esm",
}
`;
